datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -605,22 +605,23 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
     List<String> newConnectorList = new ArrayList<>(currentAssignment.keySet());
 
     List<String> deactivated = new ArrayList<>(oldConnectorList);
-    deactivated.removeAll(newConnectorList);
-    List<Future<Boolean>> assignmentChangeFutures = deactivated.stream()
-        .map(connectorType -> dispatchAssignmentChangeIfNeeded(connectorType, new ArrayList<>(), isDatastreamUpdate, true))
-        .filter(Objects::nonNull)
-        .collect(Collectors.toList());
+    List<Future<Boolean>> assignmentChangeFutures = new ArrayList<>();
+    try {
+      deactivated.removeAll(newConnectorList);
+      assignmentChangeFutures = deactivated.stream()
+          .map(connectorType -> dispatchAssignmentChangeIfNeeded(connectorType, new ArrayList<>(), isDatastreamUpdate, true))
```

Comment:
One thing to understand is, for the exception in `initializeTask`, it is possible that there is one old datastream or one datastream having issue. Based on this logic, even if one task is bad, or the host cannot process one task, the other tasks will also be not attempted. You may want to check the initializeTask loop as well.

Reply:
Basically, error in the addition of one task, should not impact the creation/removal of other tasks. So, the next retry will only attempt the task it was not able to start in the previous attempt.

Reply:
Got it, that makes sense. So basically its ok to process the new assignments partially and it doesn't have to be all or nothing. But even in that case, if even a single task processing fails we still need to queue in another handle assignment event correct ?